### ISSUE TITLE
gpt-oss sft megatron does not support sequence packing

### ISSUE THREAD

**Issue by jordane95** [NVIDIA-NeMo/RL#1685]
**Describe the bug**

When set `sequence_packing=true` and use Megatron backend for fine-tuning gpt-oss. The debug info says no attention backend available.
If `sequence_packing=False`, only `UnfusedAttention` is available, `FlashAttention` disabled for softmax_type = learnable, `FusedAttention` disabled as no backend supports the provided input.

This is different from the fine-tuning in megatron-bridge, which support `FusedAttention` with learnable softmax.

Update: The different backends enabled are due to different versions of cuDNN in docker image. Latest nemo-rl nano image has cuDNN version = 91002, but in nemo image cuDNN version is 91310.

**Steps/Code to reproduce bug**

Please list *minimal* steps or code snippet for us to be able to reproduce the bug.

A helpful guide on on how to craft a minimal bug report http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports.

**Expected behavior**

**Additional context**

**Comment by guyueh1** [comment]
@jordane95 Yes we are aware that the latest nemo-rl container can't use fused attention for gpt-oss because the cuDNN version is too old; We have a way to manually bump the cudnn version in the container so that you can use fused attention, it incurs no change to nemo-rl code but you need to install a newer cuDNN and properly expose it to LD library path. @seonjinn can share the details.

If you don't like the workaround, you can expect the newer cuDNN version will come when we bump to a newer torch version, but that might take a while.

These are all for non sequence-packing case. For sequence packing case, we don't have fused attention kernels in cudnn, so even if you bump cudnn to the latest it will still use unfused attention.

**Comment by jordane95** [comment]
> [@jordane95](https://github.com/jordane95) Hi, this issue is caused by an older cuDNN version that prevents attention fusion in the Megatron backend. You can use the sj/gpt-oss-cudnn branch as a temporary fix, which upgrades Transformer Engine to 2.10.0 and cuDNN to 9.17.0. We'll fix this permanently when we upgrade PyTorch and cuDNN in the main branch.

Thanks @seonjinn , is there any docker image which supports the new version. The latest nemo-rl docker image has lower version.

**Comment by guyueh1** [comment]
@seonjinn perhaps you can create a dockerfile that instructs how to install latest cuDNN from an existing nemo-rl container, or instructions to do that in a shell script

**Comment by dhineshkumar-r** [comment]
Hey @jordane95,
I'm looking into finetuning gpt-oss with megatron backend as well. I want to confirm the container you are using. Is it [v0.4.0](https://github.com/NVIDIA-NeMo/RL/releases/tag/v0.4.0)?
